server/utils.py

```diff
@@ -266,6 +268,46 @@ def get_dataone_package_url(repository, pid):
         return str('https://dev.nceas.ucsb.edu/#view/'+pid)
 
 
+
+def parse_jwt(jwt_token):
+    """
+    Takes a jwt token and returns the decoded section of it
+
+    :param jwt_token: The jwt token
+    :type jwt_token: str
+    :return: A dictionary of the jwt information
+    :rtype: dict
+    """
+    base_section = re.search('\.([^\.]+)\.', jwt_token).group(1)
+    if base_section is not None:
+        pad = len(base_section) % 4
+        base_section += "=" * pad
+        return str(base64.b64decode(base_section))
+    return base_section
+
+
+def extract_orcid_id(jwt):
+    """
+    Takes a JWT and extracts the orcid id out.
+    :param jwt:
+    :return:
+    :rtype: str
+    """
+    parsed_jwt = parse_jwt(jwt)
+    orcid_res = re.search('"userId"\s*:\s*"([^"]*)"', parsed_jwt).group(1)
```

Comment:
Extracting the userId with a regex seemed kinda odd to me. I thought parse_jwt returns a Dict which would lead me to thinking `extract_orcid_id` would just be implemented as:

```py
return parse_jwt(jwt).userId
```

Why the regex?

Reply:
Wanna just go ahead and use PyJWT here?